scheduler/server/job_state.go

```diff
@@ -170,12 +201,16 @@ func (j *jobState) taskCompleted(taskId string, running bool) {
 		delete(j.NotStarted, taskId)
```

Comment:
That TODO on the prev line is worrying...

Reply:
thanks for catching that.  I added it last year when I was starting to work with the (earlier) new algorithm and wasn't really sure if we needed to trap for error or what..... since then I realized we don't need it.